merge of phys-interp-screen

andrea-ferretti opened the pull request:
W-average for metallic screening.
The merge can be done, after the following issues are addressed:

- Some of the newly added files (rim_integrate_w.F, QP_interpolate_W.F) are huge, and a cleanup is definitely needed, I've left some comments in the headers of the files.
- One needs to decide whether the graphene-specific screening is included in the main release or not (personally I would leave it in a separate branch)
- Some refs in the test suite need to be updated (a phys-interp-screen branch should be created with the updated refs)

andrea-ferretti committed:
merge of devel-interpolate-screening

andrea-ferretti committed:
more files from the merge of devel-interpolate-screening

daniele-varsano commented:
*I agree clean up is needed. 
*Graphene specific I agree can be left out
*I cannot see the comments in the headers

andrea-ferretti committed:
Merge branch '5.4' into phys-interp-screen

giacomosesti committed:
Minor changes of precision
Added a comment

giacomosesti committed:
Some cleaning and code length shortening

giacomosesti committed:
Missing end do

giacomosesti committed:
wrongly cancelled variable assignement

giacomosesti committed:
Some cleaning, break part rimw_check procedure. Solved Xmat assignement issue in src/qp/QP_interpolate_W.F of no consequence, involved next nearest neighbours values not used

giacomosesti committed:
explicit indication of precision

giacomosesti committed:
subroutine splitted from QP_interpolate_W.F with check of interpolation procedure

andrea-ferretti committed:
gpu bug fixed when computing off-diagonal matrix elements of MPA sgm

giacomosesti committed:
changed YAMBO_ALLOC to ALLOCATE within OMP loops in QP_mpa.F

giacomosesti committed:
Wrong assignement of the wings indices in rim_integrate_w.F

giacomosesti committed:
replaced wrong operation sign in src/qp/QP_mpa.F

giacomosesti committed:
Removed error signal of MPA with DmRnge(0)=0.0, changed it to a warning that applies only in case of metals. \n In this latter case  DmRnge(0) is adjusted to 0.0001

nicspalla committed:
In case of Q_parallelism in the self-energy, the MPA-specific
MPI parallelism is turned off like the G_parallelism.

giacomosesti committed:
changed G_mprop calling in input

giacomosesti committed:
Merge branch 'phys-interp-screen' of github.com:yambo-code/yambo into phys-interp-screen

giacomosesti committed:
remove activation barrier for GDamping and G_mprop if DysSolver=g

andrea-ferretti committed:
Merge branch '5.4' into phys-interp-screen

andrea-ferretti committed:
naming of the output files fixed

andrea-ferretti committed:
Merge branch '5.4' into phys-interp-screen

giacomosesti committed:
Created output database for auxiliary functions activated with rimw_check
Commented debug subroutine QP_check_interpolate_W.F (still present, to be moved out)

giacomosesti committed:
Merge branch 'phys-interp-screen' of https://github.com/yambo-code/yambo into phys-interp-screen

giacomosesti committed:
Precision type error

andrea-ferretti committed:
cleanup of the sources (to be continued)

andrea-ferretti committed:
GPU fix of RIM_W based QP

andrea-ferretti committed:
Merge branch 'phys-interp-screen' of github.com:yambo-code/yambo into phys-interp-screen

sangallidavide commented:
Ciao, is there any test in the test-suite connected to this PR ?

andrea-ferretti commented:
Giacomo Sesti is working to include the tests in the test suite (indeed, due).
